Jenkinsfile

```diff
@@ -342,13 +354,22 @@ class PythonParallelTaskGenerator extends DefaultParallelTaskGenerator {
                 defaultExcludes: false
               )
             }
-            steps.env.PYTHON_VERSION = "${x}"
-            steps.env.WEBFRAMEWORK = "${y}"
-            steps.codecov(repo: "${steps.env.REPO}",
-              basedir: "${steps.env.BASE_DIR}",
-              flags: "-e PYTHON_VERSION,WEBFRAMEWORK",
-              secret: "${steps.env.CODECOV_SECRET}"
-            )
+            // steps.env.PYTHON_VERSION = "${x}"
+            // steps.env.WEBFRAMEWORK = "${y}"
+            steps.dir("${steps.env.BASE_DIR}"){
+              steps.script {
+                def massaged_py_ver = steps.sh(returnStdout: true,
+                script: """
+                    docker run apm-agent-python:${x} python -c \"import platform; pv=platform.python_version_tuple(); print('pypy' + ('' if pv[0] == 2 else str(pv[0])) if platform.python_implementation() == 'PyPy' else '.'.join(map(str, platform.python_version_tuple()[:2])))\" | egrep -v 'Starting with'
+                    """
+                ).trim()
+                steps.stash(
+                name: "coverage-${massaged_py_ver}-${y}",
+                includes: ".coverage.${massaged_py_ver}.${y}",
+                allowEmpty: false
+              )
```

Comment:
For readability reasons, I'd see it convenient to indent more the steps.stash method's arguments and closing parenthesis